Nodejs/Product/Nodejs/SharedProject/Wpf/NotificationTextBox.cs

```diff
@@ -0,0 +1,103 @@
+
+            return this.notificationTextBoxAutomationPeer;
+        }
+
+        public void RaiseNotificationEvent(string notificationText, string notificationGuid)
+        {
+            // Only attempt to raise the event if we already have an AutomationPeer.
+            if (this.notificationTextBoxAutomationPeer != null)
+            {
+                this.notificationTextBoxAutomationPeer.RaiseNotificationEvent(notificationText, notificationGuid);
+            }
+        }
+    }
+
+    internal class NotificationTextBoxAutomationPeer : TextBoxAutomationPeer
+    {
+        private NotificationTextBox notificationTextBox;
+
+        // The UIA Notification event requires the IRawElementProviderSimple
+        // associated with this AutomationPeer.
+        private IRawElementProviderSimple rawElementProviderSimple;
+
+        public NotificationTextBoxAutomationPeer(NotificationTextBox owner) : base(owner)
+        {
+            this.notificationTextBox = owner;
+        }
+
+        public void RaiseNotificationEvent(string notificationText, string notificationGuid)
+        {
+            // If we already know that the UIA Notification event is not available, do not
+            // attempt to raise it.
+            if (this.notificationTextBox.NotificationEventAvailable)
+            {
+                // If no UIA clients are listening for events, don't bother raising one.
+                if (UiaAutomationNativeMethods.UiaClientsAreListening())
+                {
+                    // Get the IRawElementProviderSimple for this AutomationPeer if we don't
+                    // have it already.
+                    if (this.rawElementProviderSimple == null)
+                    {
+                        var automationPeer = FromElement(this.notificationTextBox);
+                        if (automationPeer != null)
+                        {
+                            this.rawElementProviderSimple = this.ProviderFromPeer(automationPeer);
+                        }
+                    }
+
+                    if (this.rawElementProviderSimple != null)
+                    {
+                        try
+                        {
+                            UiaAutomationNativeMethods.UiaRaiseNotificationEvent(
+                                this.rawElementProviderSimple,
+                                UiaAutomationNativeMethods.AutomationNotificationKind.ActionCompleted,
+                                UiaAutomationNativeMethods.AutomationNotificationProcessing.All,
+                                notificationText,
+                                notificationGuid);
+                        }
+                        catch (EntryPointNotFoundException)
+                        {
+                            // The UIA Notification event is not not available, so don't attempt
+                            // to raise it again.
+                            this.notificationTextBox.NotificationEventAvailable = false;
```

Comment:
Will this happen if narrator is inactive?  What if it's activated later in the same session?

Reply:
No, I believe this should happen when clr is unable to load the UIAutomationCore.dll because the method is not found. It's a fail safe.